.github/workflows/templates-go.yaml

```diff
@@ -0,0 +1,112 @@
+name: Templates
+
+on:
+  push:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/templates-go.yaml"
+      - "templates/**"
+  pull_request:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/templates-go.yaml"
+      - "templates/**"
+
+env:
+  TINYGO_VERSION: "0.33.0"
+  GOLANGCI_VERSION: "v1.61"
+  WASH_VERSION: "0.37.0"
+  WASM_TOOLS_VERSION: "1.220.0"
+  WRPC_VERSION: "0.11.1"
+
+permissions:
+  contents: read
+
+jobs:
+  providers:
+    # Context: https://github.com/golangci/golangci-lint-action/blob/v6.1.1/README.md#annotations
+    permissions:
+      # Required: allow read access to the content for analysis.
+      contents: read
+      # Optional: allow write access to checks to allow the action to annotate code in the PR.
+      checks: write
+    strategy:
+      matrix:
+        template-name:
+          - keyvalue-inmemory
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: taiki-e/install-action@a2ff97ae7a403b9e691f3d5d5e214415068a681b # v2.46.7
+        with:
+          tool: wash-cli@${{ env.WASH_VERSION }}
+      - uses: taiki-e/install-action@a2ff97ae7a403b9e691f3d5d5e214415068a681b # v2.46.7
+        with:
+          tool: wasm-tools@${{ env.WASM_TOOLS_VERSION }}
+      - uses: taiki-e/install-action@a2ff97ae7a403b9e691f3d5d5e214415068a681b # v2.46.7
+        with:
+          tool: wrpc@${{ env.WRPC_VERSION }}
```

Comment:
If you feel like it, you could also combine these into a single action, something like:
```
      - uses: taiki-e/install-action@a2ff97ae7a403b9e691f3d5d5e214415068a681b # v2.46.7
        with:
          tool: wash-cli@${{ env.WASH_VERSION }},wasm-tools@${{ env.WASM_TOOLS_VERSION }},wrpc@${{ env.WRPC_VERSION }}
```

Reply:
so 🍿 
I had that before ( copied from another workflow ), and seemed like it wasnt properly caching installations by having all in the same `tool`.

now re-reading the latests commits I think it was `wrpc` compilation all along.
I'm combining them into a single action again and think we should look into adding our tools here like wasmtime https://github.com/taiki-e/install-action/blob/main/tools/codegen/base/wasmtime.json